responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -370,6 +376,14 @@ def counterfactual(self) -> CounterfactualManager:
         """
         return self._counterfactual_manager
 
+    @property
+    def data_balance(self) -> DataBalanceManager:
```

Comment:
I just checked with pm
It is probably better to always have data balance instead of require user to call another add method.

Reply:
How computationally intensive are the calculations?

Reply:
Tagging @imatiach-msft and @PIC123 also.

Hi guys, I ran a small benchmark, and calculating measures is not very compute intensive, especially for datasets with <1M rows.
I ran these tests on my local machine (AMD Ryzen 5 3600 Processor, 16 GB RAM, M.2 SSD, has a GPU but not applicable):
```
Running data balance measures on 7 columns
Time taken for dataset with 48,842 rows: 1.5469965934753418 seconds
Time taken for dataset with 97,684 rows: 1.3919968605041504 seconds
Time taken for dataset with 195,368 rows: 1.6330010890960693 seconds
Time taken for dataset with 390,736 rows: 1.9849941730499268 seconds
Time taken for dataset with 781,472 rows: 2.7039999961853027 seconds
Time taken for dataset with 1,562,944 rows: 4.72900128364563 seconds
Time taken for dataset with 3,125,888 rows: 7.672998905181885 seconds
Time taken for dataset with 6,251,776 rows: 14.046992540359497 seconds
Time taken for dataset with 12,503,552 rows: 26.853999853134155 seconds
Time taken for dataset with 25,007,104 rows: 75.52636933326721 seconds
```

---

Pasting my script for future references:

```
import time
from responsibleai.managers.data_balance_manager import DataBalanceManager


# from https://publictestdatasets.blob.core.windows.net/data/adult-train.csv and adult-test.csv
train_df, test_df = train_data, test_data 
target_feature = 'income'
categorical_features = ['education', 'marital-status',
                        'occupation', 'relationship', 'race', 'gender', 'native-country']

print(f"Running data balance measures on {len(categorical_features)} columns")

for i in range(10):
    t0 = time.time()

    manager = DataBalanceManager(
        train=train_df,
        test=test_df,
        target_column=target_feature,
        task_type="classification",
    )
    manager.add(cols_of_interest=categorical_features)
    manager.compute()
    
    t1 = time.time()
    num_rows = manager._df.shape[0]
    print(f"Time taken for dataset with {num_rows:,} rows: {t1 - t0} seconds")

    # Double the size of each dataframe
    train_df = pd.concat([train_df, train_df])
    test_df = pd.concat([test_df, test_df])
```

Reply:
@imatiach-msft @romanlutz @memoryz @martthalch 

Calling out two big changes that I discussed with @xuke444 and @PIC123:

### Change 1

Instead of the user adding data balance the same way they would for managers:
```
rai_insights = RAIInsights(...)
rai_insights.data_balance.add(cols_of_interest=["col1", "col2"])
rai_insights.compute()
```

We moved towards adding data balance when user creates `RAIInsights`. Compute still needs to be called.
```
rai_insights = RAIInsights(...)
rai_insights.compute()
```

If the following criteria is met:
1. `task_type == "classification"`
2. `len(categorical_features) > 0`
3. target column `classes` is not null

Then data balance measures will be automatically added for the given `categorical_features` list.

Also, user no longer has access to `rai_insights.data_balance` property
But they can still set their own cols_of_interest via
```
rai_insights._data_balance_manager.add(...)  <-- overwrites the automatic add from RAIInsights constructor
```

---

### Change 2

Initially, the user was required to pass in the positive label for their dataset. For example, if the target column contains {<=50k, >50k}, feature balance measures need to know which of these is considered the positive label to compute measures.

Now, feature balance measures are being computed for all values in the target column. That way, user can choose in the UI which label is considered the positive label.

![image](https://user-images.githubusercontent.com/64443771/178053324-ab40c5a5-c12d-438b-b5d6-6d496adfdcfc.png)

This also makes it so that data balance can work for all classification scenarios (both binary and multiclass) instead of just binary like before. 😁